### ISSUE TITLE
SwUpdate: versionUpdates and available not firing

### ISSUE THREAD

**Issue by phrei** [angular/angular#45707]
### Which @angular/* package(s) are the source of the bug?

service-worker

### Is this a regression?

Yes

### Description

After updating to Angular 13 from Angular 12, I cannot detect updates from the Service Worker anymore.

I had the following code which used to work fine:

    import { Injectable } from "@angular/core";
    import { SwUpdate } from "@angular/service-worker";
    @Injectable()
    export class UpdateService {
        constructor(updates: SwUpdate) {
            updates.available.subscribe(async event => {
                await updates.activateUpdate();
                location.reload();
            });
        }
    }

It didn't work anymore (although it should as far as I know). According to the [docs](https://angular.io/guide/service-worker-communications) I've changed my code to:

    import { Injectable } from "@angular/core";
    import { SwUpdate } from "@angular/service-worker";
    import { interval } from "rxjs";
    @Injectable()
    export class UpdateService {
        constructor(updates: SwUpdate) {
            console.log('UpdateService: Constructor', updates.isEnabled);

            // This shouldn't be necessary but is a try to get the versionUpdates. Doesn't do it either.
            interval(10000).subscribe(() => {
                console.log('UpdateService: Checking for Updates')
                updates.checkForUpdate();
            });

            updates.versionUpdates.subscribe(async evt => {
                console.log('UpdateService: versionUpdates', evt);
                switch (evt.type) {
                    case 'VERSION_DETECTED':
                        console.log(`Downloading new app version: ${evt.version.hash}`);
                        break;
                    case 'VERSION_READY':
                        console.log(`Current app version: ${evt.currentVersion.hash}`);
                        console.log(`New app version ready for use: ${evt.latestVersion.hash}`);
                        await updates.activateUpdate();
                        location.reload();
                        break;    
                    case 'VERSION_INSTALLATION_FAILED':
                        console.log(`Failed to install app version '${evt.version.hash}': ${evt.error}`);
                        break;
                }
            });
        }
    }

In my console I see that the constructor is being called and that it checks for updates every 10 seconds (according to my interval). Also I see in the network tab, that new versions of the main.js is being downloaded as I update my app - just my subscription to the versionUpdates observable is never called.

### Please provide the environment you discovered this bug in (run `ng version`)

```
Angular CLI: 13.3.3
Node: 14.17.0
Package Manager: npm 7.15.1
OS: win32 x64

Angular: 13.3.4
... animations, cdk, common, compiler, compiler-cli, core, forms
... material, platform-browser, platform-browser-dynamic, router
... service-worker

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1303.3
@angular-devkit/build-angular   13.3.3
@angular-devkit/core            13.3.3
@angular-devkit/schematics      13.3.3
@angular/cli                    13.3.3
@angular/fire                   7.3.0
@angular/flex-layout            13.0.0-beta.38
@schematics/angular             13.3.3
rxjs                            7.5.5
typescript                      4.6.3
```

**Comment by phrei** [comment]
@gkalpak I'm having a hard time providing you with a project (as you would need to have access to the code and be able to publish it so you can see the effects). Also if I create a minimal project, it cannot be run locally as the ngsw only works in production.

My best bet would be: Create a vanilla Angular Project and add @angular/pwa according to https://angular.io/guide/service-worker-getting-started and publish it some place and create a service like the one I described and see if console-messages appear when you update and redeploy your code.

I couldn't make a Stackblitz to show the effects either.

**Comment by gkalpak** [comment]
You don't have to publish the project. It _is_ possible to run the Angular app locally in production mode and get the SW to work (just not with `ng serve`).

**Comment by phrei** [comment]
I was able to fix the issue: The `angular.json` was configured to only add the serviceworker (with the `ngswConfigPath`) when the "production" configuration was used. In some update I removed the `--prod`-Option from `ng build` which made the Service-Worker-File (`ngsw-worker.js`) disappear. Because I didn't delete, only replace, files when publishing my project, the `ngsw-worker.js` was still there from previous versions and so I haven't noticed the missing file in production for some time.

If a new angular-project is generated with the `@angular/pwa`-Command, the following properties are configured for "build" - regardless of which configuration (production or development) is used:

    "serviceWorker": true,
    "ngswConfigPath": "src/ngsw-config.json"
